rosbridge_library/test/internal/test_message_conversion.py

```diff
@@ -175,40 +185,22 @@ def test_string_base_msg(self):
             self.do_primitive_test(x, "std_msgs/String")
 
     def test_time_msg(self):
-        msg = {"data": {"secs": 3, "nsecs": 5}}
-        self.do_test(msg, "std_msgs/Time")
+        msg = {"sec": 3, "nanosec": 5}
+        self.do_test(msg, "builtin_interfaces/Time")
 
-        msg = {"times": [{"secs": 3, "nsecs": 5}, {"secs": 2, "nsecs": 7}]}
+        msg = {"times": [{"sec": 3, "nanosec": 5}, {"sec": 2, "nanosec": 7}]}
```

Comment:
Does this mean the JSON format of `time`/`duration` fields is changing for ROS 2? Will clients need to update their code? I think we need to maintain backward compatibility with the `secs`/`nsecs` format if that is the format existing JSON clients are expecting.

Reply:
> Does this mean the JSON format of time/duration fields is changing for ROS 2?

As a test spec, yes.
As an application spec, no. It's already changed.

To test this, please run the following commands.

Case 1. Working
```sh-session
// Terminal 1
$ ros2 run rosbridge_server rosbridge_websocket
// normal outputs

// Terminal 2
$ curl -sL https://github.com/tier4/rosbridge_suite/raw/add-test-scripts/test_scripts/time.py | python
publish: {'sec': 1, 'nanosec': 2}
subscribe: {'sec': 1, 'nanosec': 2}
publish: {'sec': 1, 'nanosec': 2}
subscribe: {'sec': 1, 'nanosec': 2}
```

Case 2. Not working
```sh-session
// Terminal 1
$ ros2 run rosbridge_server rosbridge_websocket
[ERROR 1634483257.554689935] [rosbridge_websocket]: [Client 4] [id: publish:/time:3] publish: 'Time' object has no attribute 'secs'
[ERROR 1634483258.559444873] [rosbridge_websocket]: [Client 4] [id: publish:/time:4] publish: 'Time' object has no attribute 'secs'

// Terminal 2
$ curl -sL https://github.com/tier4/rosbridge_suite/raw/add-test-scripts/test_scripts/time.py | python
publish: {'secs': 1, 'nsecs': 2}
publish: {'secs': 1, 'nsecs': 2}
```

Reply:
Even though it's already changed, we could still choose to change it back (and we should bump the major version of the package if it's not backward compatible). Do you think users of the library will be confused by the difference in behavior between ROS 1/2? If someone upgrades from ROS 1 to ROS 2 their websocket client will start receiving values. Should we change it back to `secs/nsecs` or at least change the publish side to accept both forms?

cc @defunctzombie @amacneil for thoughts.

Reply:
@jtbandes I'll leave the decision to the maintainers, but my personal (and selfish) thought is, since our company has completely moved to ROS2, either one is fine.

> Do you think users of the library will be confused by the difference in behavior between ROS 1/2?

Yes, I remember we had to fix some web application code for ROS2, but it wasn't so big confusion.

But anyway, I think clients should change the data format when they receive the data because `rosbridge` outputs the same data for each client. Is this correct or does `rosbridge` change output data depending on the client ROS version?

Reply:
I'm not following the comments in this thread. Is the statement that rosbridge _currently_ outputs `sec/nsec` and this is confusing for ros2 users and we should change to sec/nanosec? Or that rosbridge outputs `sec/nanosec` and this is confusing for ros1 users?

What is the specific change (if any) being requested of the api/protocol?

Reply:
Currently it outputs secs/nsecs for ROS 1 and sec/nanosec for ROS 2. This PR just changes the test.

Reply:
If this is just changing a test then we should merge the PR to have tests and tackle API changes in separate discussions.